server/src/main/scala/viper/gobraserver/backend/ViperServer.scala

```diff
@@ -39,6 +39,15 @@ object ViperServer {
           case e: Throwable => verificationPromise tryFailure e
         }
 
+      case Status.Success =>
+        // this message is the last one to be received meaning that all messages belonging to the current verification
+        // should be received by now. To make sure that the promise is eventually completed (even if not
+        // OverallSuccessMessage or OverallFailureMessage was received), we let the promise fail. This failure is
+        // ignored if the promise has already been completed before:
```

Comment:
Is there any particular reason to make the promise fail instead of succeeding (via a call to `trySuccess`)?

Reply:
We would need to provide a result of type `VerificationResult`. However, if the promise is not completed yet, then our assumption that there always is an OverallSuccess or OverallFailure message is violated. This case here simply ensures that there is no non-termination because of this violation